File: src/ruby/seqlength.rb
#!/usr/bin/env ruby
 
$/ = ">"
ARGF.gets
while rec = ARGF.gets
  rec.chomp!
  nl = rec.index("\n")
  header = rec[0..nl-1]
  seq = rec[nl+1..-1]
  seq.gsub!(/\n/,'')
  puts [header, seq.length].join(" ")
end

File: src/python/seqlength.py
#!/usr/bin/env python
 
import os,sys
 
f = open(sys.argv[1],'rU')
header = f.readline()
header = header.rstrip(os.linesep)
sequence=''
for line in f:
  line = line.rstrip('\n')
  if(line[0] == '>'):
    header = header[1:]
    header = line
    print header, len(sequence)
    sequence = ''
  else:
    sequence += line
 
print header, len(sequence)